### PR TITLE
Switch to zip

### PR DESCRIPTION
**Pull request by nmuesch**
### Requirements for Contributing to this repository

* Fill out the template below. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* The pull request must only fix one issue, or add one feature, at the time.

### What does this PR do?

Add support for windows cloudfoundry buildpack environments
Supersedes - Install windows agent on windows containers

* Adds a datadog.bat file to easily interact with the agent executable to run things like the status command without needing to point to the config file
* Adds the .NET tracer and documents the way to automatically trace .NET applications
* Enables system metrics by default (including running dogstatsd)

(Uses the agent binaries instead of the full package)

### Description of the Change

Uses the new zip artifact and starts the process of running single binaries as opposed to the full Datadog Agent

### Additional Notes

Currently this starts the puppy agent and the trace agent.
Since we no longer can rely on the Agent's python, the few python scripts that do setup work still need to be ported over. 

### Review checklist (to be filled by reviewers)

- [ ] PR title must be written as a CHANGELOG entry [(see why)](/CONTRIBUTING.md#pull-request-title)
- [ ] Files changes must correspond to the primary purpose of the PR as described in the title (small unrelated changes should have their own PR)
- [ ] PR must have one `changelog/` label attached. If applicable it should have the `backward-incompatible` label attached.
- [ ] PR should not have `do-not-merge/` label attached.
- [ ] If Applicable, issue must have `kind/` and `severity/` labels attached at least.

### CODE REVIEW

File: bin/supply.ps1 [hunk 1]
```diff
@@ -3,37 +3,30 @@ echo "-----> DatadogBuildpack/supply"
...
+# $ENV_DIR=$ARGS[2]
 
-echo "       Installing Datadog Agent"
+echo "       Extracting Datadog Agent"
 
 
 New-Item -Path "$BUILD_DIR\datadog\scripts" -ItemType Directory
-New-Item -Path "$BUILD_DIR\datadog\extracted-agent" -ItemType Directory
 New-Item -Path "$BUILD_DIR\.profile.d" -ItemType Directory
+New-Item -Path "$BUILD_DIR\datadog/AppData" -ItemType Directory
```
Comment (zippolyte) [inline bin/supply.ps1]:
```suggestion
New-Item -Path "$BUILD_DIR\datadog\AppData" -ItemType Directory
```

File: bin/supply.ps1 [hunk 2]
```diff
@@ -3,37 +3,30 @@ echo "-----> DatadogBuildpack/supply"
...
 
 # Extract agent
-Start-Process "MSIEXEC" -ArgumentList "/a $ROOT_DIR\lib\datadog-agent.msi /qn TARGETDIR=$BUILD_DIR\datadog\extracted-agent" -Wait -NoNewWindow
-Move-Item -Path "$BUILD_DIR\datadog\extracted-agent\Datadog\Datadog Agent" -Destination "$BUILD_DIR\datadog\"
-Move-Item -Path "$BUILD_DIR\datadog\extracted-agent\CommonAppData\Datadog" -Destination "$BUILD_DIR\datadog\AppData"
-Remove-Item -Force -Recurse -Path "$BUILD_DIR\datadog\extracted-agent"
+Expand-Archive -LiteralPath $ROOT_DIR\lib\cf-windows-7.17.0-buildpack.zip -DestinationPath $BUILD_DIR\datadog
```
Comment (zippolyte) [inline bin/supply.ps1]: Can we have a versionless zip name ?

Comment (nmuesch) [reply bin/supply.ps1]: Yeah for sure. This part isn't final yet until the build script is updated to pull the artifact from somewhere. Waiting on an agent PR before I can add that 😄

**Comment by nmuesch** [comment]
Closing this in favor of - https://github.com/DataDog/datadog-cloudfoundry-buildpack/pull/82

**Pull request closed without merging**